src/test/scala/org/apache/spark/deploy/armada/e2e/TestOrchestrator.scala

```diff
@@ -144,37 +183,52 @@ class TestOrchestrator(
     import scala.concurrent.duration._
 
     // Give Armada some time to schedule resources after driver starts (executors need time to come up)
-    val maxWaitTime                 = 30.seconds
-    val startTime                   = System.currentTimeMillis()
-    var attempts                    = 0
-    var lastResult: AssertionResult = AssertionResult.Failure("Not yet checked")
+    val maxWaitTime                    = 30.seconds
+    val gracePeriodAfterCompletion     = 10.seconds
```

Comment:
In client mode tests, the following error often came up.
```
 - Executor labels: Executor labels: Job completed before assertion could pass
  - Node selector verification: Node selector verification: Job completed before assertion could pass
[DEBUG] Capturing debug info for namespace e2e-test-a69f6cde
[DEBUG] Capturing logs for pod armada-01kd6k0v5653r4mftqz9cm449r-0
[DEBUG] Capturing logs for pod armada-01kd6k0v5653r4mftqzarty2v5-0
```
This is because validateScheduling exists as soon as `jobCompleted()` is true, so in client mode the job can sometimes finish before assertions are run. Therefore, causing the above issue.

The grace period allows for continuing to run assertions even after the job completes.